src/components/Publish/Actions/index.tsx

```diff
@@ -55,10 +61,28 @@ export default function Actions({
     handleAction('prev')
   }
 
-  const isContinueDisabled =
-    (values.user.stepCurrent === 1 && errors.metadata !== undefined) ||
-    (values.user.stepCurrent === 2 && errors.services !== undefined) ||
-    (values.user.stepCurrent === 3 && errors.pricing !== undefined)
+  useEffect(() => {
+    setContinueDisabled(
+      (values.user.stepCurrent === 1 && errors.metadata !== undefined) ||
+        (values.user.stepCurrent === 2 && errors.services !== undefined) ||
+        (values.user.stepCurrent === 3 && errors.pricing !== undefined)
+    )
+  }, [errors, values.user.stepCurrent])
+
+  useEffect(() => {
```

Comment:
also this condition might get more complex as I'm not sure about throwing error when no account connected, as this could only apply for dynamic pricing so splitting this up into the components where it's relevant and from there fire `setFieldError()` should also help in not having this become too complex of a condition.

No clue why we have our own error state management in the `Pricing/Dynamic.tsx` component but as we have the respective checks there already, some addition like this would be compromise as to not completely refactor error display in that component:

```tsx
// Check: account, network & insufficient balance
  useEffect(() => {
    if (!accountId) {
      setError(`No account connected. Please connect your Web3 wallet.`)
      setFieldError(
        'pricing',
        'No account connected. Please connect your Web3 wallet.'
      )
    } else if (Number(balance.ocean) < Number(amountOcean)) {
      setError(`Insufficient balance. You need at least ${amountOcean} OCEAN.`)
      setFieldError(
        'pricing',
        `Insufficient balance. You need at least ${amountOcean} OCEAN.`
      )
    } else {
      setError(undefined)
      setFieldError('pricing', undefined)
    }
  }, [amountOcean, networkId, accountId, balance])
```

Then a potential later refactor could remove the need for `setError()` or maybe out of the box `setFieldError()` in there would produce already another error area

Reply:
See, trust the Formik state :-) 

<img width="929" alt="Screen Shot 2022-05-17 at 18 32 45" src="https://user-images.githubusercontent.com/90316/168875115-2a054ab4-f442-4464-9aa6-f64cd24fb85f.png">

We can now remove that `setError('Insufficient balance..')` line which is responsible for showing the below error alert, whereas `setFieldError()` gives uses a nice user-friendly placement of the error

Reply:
actually made sense moving this custom error state in Dynamic pricing component, as this was greatly out of scope with the issue fix, I'll quickly push what I have.

But you're right that I find this `setFieldError()` to be sluggish at times, but this might be because we dadditionally have lots of custom validation methods we are running over the fields in there too.

Reply:
well, giving up for today cause there's the theory of the Formik docs and then there's reality 🙃

- https://github.com/jaredpalmer/formik/issues/1278
- https://github.com/jaredpalmer/formik/issues/706
- https://github.com/jaredpalmer/formik/issues/1111

But definetly now we know where exactly to show those errors. And then this was my next thinking too to get this current setup working as intended: https://github.com/jaredpalmer/formik/issues/706#issuecomment-583834723

As all those different errors should show up on the same field instead of being plastered around the screen:

<img width="478" alt="Screen Shot 2022-05-17 at 19 21 04" src="https://user-images.githubusercontent.com/90316/168883567-ecfea677-b975-41b3-9af0-fab7082b6753.png">

<img width="924" alt="Screen Shot 2022-05-17 at 19 00 06" src="https://user-images.githubusercontent.com/90316/168883586-c2d746eb-08ad-4e48-8831-0f7df61f5ff4.png">

<img width="715" alt="Screen Shot 2022-05-17 at 19 42 00" src="https://user-images.githubusercontent.com/90316/168886796-a0004d57-4a33-4e8b-bd20-1b25229a0e88.png">

Reply:
> But you're right that I find this `setFieldError()` to be sluggish at times, but this might be because we dadditionally have lots of custom validation methods we are running over the fields in there too.

Yeah, unfortunately using the `setFieldError` alone introduces some weird side effects. For some reason our `setFieldError` just gets overwritten by some state refresh I haven't identify yet. 

https://user-images.githubusercontent.com/13741335/169017655-c8194180-206e-49b8-8ffc-245530d1f5a2.mp4